### PR TITLE
accomodate oliver's new FS checks during warfile packing and installation

### LINKED ISSUE #330: automated maven warfile package fails, manual succeeds
currently using maven 3.8.8. special env vars (previously) necessary:

```
export JAVA_HOME=/usr/lib/jvm/jre-openjdk
export M2_HOME=/opt/maven
export MAVEN_HOME=/opt/maven
export PATH=${M2_HOME}/bin:${PATH}
```

### PR DESCRIPTION
**Pull request by donsizemore**
Closes #330

### CODE REVIEW

File: tasks/dataverse-build.yml [hunk 1]
```diff
@@ -32,27 +32,35 @@
     chdir: "{{ dataverse.srcdir }}"
   when: dataverse.unittests.enabled == false and
         maven.version == 'default'
+  become: true
+  become_user: '{{ dataverse.payara.user }}'
 
 - name: build warfile without tests and custom maven. tail /tmp/dataverse/mvn.out for gory details.
```
Comment (pdurbin) [inline tasks/dataverse-build.yml]:
```suggestion
- name: build warfile without tests and custom maven. tail /home/dataverse/dataverse/mvn.out for gory details.
```

File: tasks/dataverse-build.yml [hunk 2]
```diff
@@ -32,27 +32,35 @@
...
   shell: "source /etc/profile.d/maven.sh && mvn -T 2C clean package -Dmaven.test.skip=true > mvn.out"
   args:
     chdir: "{{ dataverse.srcdir }}"
   when: dataverse.unittests.enabled == false and
         maven.version != 'default'
+  become: true
+  become_user: '{{ dataverse.payara.user }}'
 
 - name: build warfile with specified tests. tail /tmp/dataverse/mvn.out for gory details.
```
Comment (pdurbin) [inline tasks/dataverse-build.yml:47]:
```suggestion
- name: build warfile with specified tests. tail /home/dataverse/dataverse/mvn.out for gory details.
```

File: tasks/dataverse-build.yml [hunk 3]
```diff
@@ -32,27 +32,35 @@
...
   shell: "mvn {{ dataverse.unittests.argument }} {{ jacoco_arg }} -T 2C clean package > mvn.out"
   args:
     chdir: "{{ dataverse.srcdir }}"
   when: dataverse.unittests.enabled == true and
         maven.version == 'default'
+  become: true
+  become_user: '{{ dataverse.payara.user }}'
 
 - name: build warfile with specified tests using custom maven. tail /tmp/dataverse/mvn.out for gory details.
```
Comment (pdurbin) [inline tasks/dataverse-build.yml]:
```suggestion
- name: build warfile with specified tests using custom maven. tail /home/dataverse/dataverse/mvn.out for gory details.
```

File: tasks/payara.yml [hunk 4]
```diff
@@ -136,16 +133,7 @@
     mode: '0750'
     state: directory
 
-- name: workaround for FISH-7665
-  lineinfile:
-    path: '{{ payara_dir }}/glassfish/config/asenv.conf'
-    line: '{{ item }}'
-  with_items:
-    - 'export JAVA_TOOL_OPTIONS'
-    - 'JAVA_TOOL_OPTIONS="-Djdk.util.zip.disableZip64ExtraFieldValidation=true --add-opens=java.base/java.io=ALL-UNNAMED"'
```
Comment (pdurbin) [inline tasks/payara.yml:145]: Huh. Is this safe to remove? We still have it in our Dockerfile:

```
$ grep UNN modules/container-base/src/main/docker/Dockerfile -B1
    # Workaround for FISH-7722: Failed to deploy war with @Stateless https://github.com/payara/Payara/issues/6337
    ${ASADMIN} create-jvm-options --add-opens=java.base/java.io=ALL-UNNAMED
```

**Review by pdurbin — COMMENTED** [review]
At a high level, makes sense. I updated /tmp/dataverse/mvn.out here and there. Wondering about the FISH-7722 thing.

File: tasks/payara.yml (hunk 4 above)
Comment (donsizemore) [reply tasks/payara.yml:145]: you're right, I thought this was fixed, but not until 6.2023.9 - will keep.

**Pull request merged**